agent/engine/task_manager.go

```diff
@@ -624,43 +678,60 @@ func (mtask *managedTask) handleContainerStoppedTransitionError(event DockerCont
 	return true
 }
 
-// progressContainers tries to step forwards all containers that are able to be
+// progressTask tries to step forwards all containers and resources that are able to be
 // transitioned in the task's current state.
 // It will continue listening to events from all channels while it does so, but
 // none of those changes will be acted upon until this set of requests to
 // docker completes.
 // Container changes may also prompt the task status to change as well.
-func (mtask *managedTask) progressContainers() {
-	seelog.Debugf("Managed task [%s]: progressing containers in task", mtask.Arn)
+func (mtask *managedTask) progressTask() {
+	seelog.Debugf("Managed task [%s]: progressing containers and resources in task", mtask.Arn)
 	// max number of transitions length to ensure writes will never block on
 	// these and if we exit early transitions can exit the goroutine and it'll
 	// get GC'd eventually
-	transitionChange := make(chan struct{}, len(mtask.Containers))
-	transitionChangeContainer := make(chan string, len(mtask.Containers))
+	transitionChange := make(chan struct{}, len(mtask.Containers)+len(mtask.Resources))
+	transitionChangeEntity := make(chan string, len(mtask.Containers)+len(mtask.Resources))
+
+	// anyResourceTransition is set to true when transition function needs to be called or
+	// known status can be changed
+	anyResourceTransition, resTransitions := mtask.startResourceTransitions(
```

Comment:
Please make a comment that the `startResourceTransitions` has to be called before `startContainerTransitions`.